File: 프로그래머스/2/84512. 모음 사전/모음 사전.py
from itertools import product

def solution(word):
    answer = 0
    dic = []
    for i in range(1, 6):
        per = product(['A', 'E', 'I', 'O', 'U'], repeat=i)
        for p in per:
            tmp = ''
            for a in p:
                tmp+=a
            dic.append(tmp)
    dic = sorted(dic)
    
    for i in range(len(dic)):
        if word==dic[i]:
            return i+1

File: 프로그래머스/2/84512. 모음 사전/모음 사전.java
import java.util.*;

class Solution {
    
    static List<String> list;
    static char[] alphabets = {'A', 'E', 'I', 'O', 'U'};
    
    public int solution(String word) {
        int answer = 0;
        list = new ArrayList<>();
        
        //1. 가능한 문자열을 모두 만듦
        makeStr("", 0);
        //2. 문자열을 정렬함
        String[] dic = list.toArray(new String[list.size()]);
        Arrays.sort(dic);
        
        
        
        for(int i=0; i<dic.length; i++){
            if(dic[i].equals(word)) return i+1;
        }
        return answer;
    }
    
    private void makeStr(String str, int len){
        if(len==5) return;
        
        for(char c: alphabets){
            if(!list.contains(str+c)){
                makeStr(str+c, len+1);
                list.add(str+c);
            }
        }
    }
    
}